workbench/src/renderer/components/SetupTab/ArgInput/index.jsx

```diff
@@ -220,30 +225,15 @@ export default function ArgInput(props) {
   let form;
   if (argSpec.type === 'boolean') {
     form = (
-      <React.Fragment>
-        <Form.Check
-          id={argkey}
-          inline
-          type="radio"
-          label="Yes"
-          value="true"
-          checked={!!value} // double bang casts undefined to false
-          onChange={handleBoolChange}
-          name={argkey}
-          disabled={!enabled}
-        />
-        <Form.Check
-          id={argkey}
-          inline
-          type="radio"
-          label="No"
-          value="false"
-          checked={!value} // undefined becomes true, that's okay
-          onChange={handleBoolChange}
-          name={argkey}
-          disabled={!enabled}
-        />
-      </React.Fragment>
+      <Form.Check
+        inline
+        type="switch"
+        id={argkey}
+        name={argkey}
+        onChange={() => updateArgValues(argkey, !value)}
+        disabled={!enabled}
+        bsCustomPrefix="form-switch"
```

Comment:
I think we need the `checked` prop here same as before. That way the state can be controlled by interactions other than just clicks, like when the app passes a set of initial arg values from a datastack file. 

And I suppose we should cover that by a test too. Sorry we didn't do it earlier! Maybe these ideas will help... In `setuptab.test.js` we could maybe add/modify tests in `describe('Arguments form input types'` to also expect a value for the input.  Something like 

```javascript
test.only('render a toggle with a value', async () => {
    const spec = baseArgsSpec('boolean');
    const { findByLabelText } = renderSetupFromSpec(spec, UI_SPEC, { arg: true });
    const input = await findByLabelText(`${spec.args.arg.name}`);
    // for some reason, the type is still checkbox when it renders as a switch
    expect(input).toHaveAttribute('type', 'checkbox');
    expect(input).toBeChecked();
  });
```
Where we added a third optional parameter to `renderSetupFromSpec`:
```javascript
function renderSetupFromSpec(baseSpec, uiSpec, initValues = undefined)
```
so we can pass a default args-dict object `{ arg: true }` to it and use it like:
```jsx
<SetupTab
      argsInitValues={initValues}
```

Reply:
Nice, thanks for catching that!

Fortunately this was as simple as adding `checked={value}`. I made the test changes pretty much exactly as you suggested.